deepmd/dpmodel/model/linear_model.py

```diff
@@ -0,0 +1,322 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+from typing import (
+    Dict,
+    List,
+    Optional,
+    Tuple,
+    Union,
+)
+
+import numpy as np
+
+from deepmd.dpmodel import (
+    FittingOutputDef,
+    OutputVariableDef,
+)
+from deepmd.dpmodel.utils.nlist import (
+    build_multiple_neighbor_list,
+    get_multiple_nlist_key,
+    nlist_distinguish_types,
+)
+
+from .base_atomic_model import (
+    BaseAtomicModel,
+)
+from .dp_atomic_model import (
+    DPAtomicModel,
+)
+from .pair_tab_model import (
+    PairTabModel,
+)
+
+
+class LinearModel(BaseAtomicModel):
+    """Linear model make linear combinations of several existing models.
+
+    Parameters
+    ----------
+    models : list[DPAtomicModel or PairTabModel]
+        A list of models to be combined. PairTabModel must be used together with a DPAtomicModel.
+    weights : list[float] or str
+        If the type is list[float], a list of weights for each model.
+        If "mean", the weights are set to be 1 / len(models).
+        If "sum", the weights are set to be 1.
+        If "zbl", the weights are calculated internally. This only allows the combination of a PairTabModel and a DPAtomicModel.
+    """
+
+    def __init__(
+        self,
+        models: List[Union[DPAtomicModel, PairTabModel]],
+        weights: Union[str, List[float]],
+        **kwargs,
+    ):
+        super().__init__()
+        self.models = models
+        self.weights = weights
+        self.distinguish_type_list = [
+            model.distinguish_types() for model in self.models
+        ]
+
+    def distinguish_types(self) -> bool:
+        """If distinguish different types by sorting."""
+        return False
+
+    def get_rcut(self) -> float:
+        """Get the cut-off radius."""
+        return max(self.get_rcuts())
+
+    def get_rcuts(self) -> List[float]:
+        """Get the cut-off radius for each individual models in ascending order."""
+        return [model.get_rcut() for model in self.models]
+
+    def get_sel(self) -> List[int]:
+        return [max(self.get_sels())]
+
+    def get_sels(self) -> List[int]:
+        """Get the cut-off radius for each individual models in ascending order."""
+        return [
+            sum(model.get_sel())
+            if isinstance(model.get_sel(), list)
+            else model.get_sel()
```

Comment:
for indistinguished type models, the get_sel method should also return List[int], the list is of length 1.